File: new/atoms/site-vocab.test.mjs
import { siteVocab as __pure_siteVocab } from './site-vocab.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הטיהור v3)
const __d_siteVocab_SITE_VOCAB_T = {
  k1: "Get in touch",
  k2: "צרו קשר",
  k3: "Contact",
  k4: "Contact us",
  k5: "Your request",
  k6: "הפנייה שלך",
  k7: "Donate now",
  k8: "לתרומה עכשיו",
  k9: "Donate",
  k10: "לתרומה",
  k11: "Your gift",
  k12: "התרומה שלך",
};
const siteVocab = (...a) => __pure_siteVocab(...a, ...Array(Math.max(0, 2 - a.length)).fill(undefined), __d_siteVocab_SITE_VOCAB_T);
const C = [
  [false, 'he', { heroCta: 'לתרומה עכשיו', navCta: 'לתרומה ♡', give: 'לתרומה ♡', giveLabel: 'התרומה שלך', commercial: false }],
  [false, 'en', { heroCta: 'Donate now', navCta: 'Donate ♡', give: 'Donate ♡', giveLabel: 'Your gift', commercial: false }],
  [true, 'he', { heroCta: 'צרו קשר', navCta: 'צרו קשר', give: 'צרו קשר', giveLabel: 'הפנייה שלך', commercial: true }],
  [true, 'en', { heroCta: 'Get in touch', navCta: 'Contact', give: 'Contact us', giveLabel: 'Your request', commercial: true }],
];
let f = 0;
for (const [commercial, lang, w] of C) {
  const g = siteVocab(commercial, lang);
  if (JSON.stringify(g) !== JSON.stringify(w)) {
    console.error(`✗ (${commercial},'${lang}') ⇒ ${JSON.stringify(g)} ≠ ${JSON.stringify(w)}`);
    f = 1;
  }
}
// דוגמה 5: יידיש נופלת לעברית — זהה-ביט ל-(false,'he')
if (JSON.stringify(siteVocab(false, 'yi')) !== JSON.stringify(siteVocab(false, 'he'))) {
  console.error("✗ (false,'yi') אינו זהה ל-(false,'he')");
  f = 1;
}
if (f) process.exit(1);
console.log('✓ site-vocab: 5 דוגמאות-חוזה — ירוק');

File: new/dart-maor/site-vocab_test.dart
// רתמת-זהב · site-vocab — כל דוגמאות-החוזה + בדיקת-ה-JS (site-vocab.test.mjs), זהות.
// השוואת-מפה = מפתחות (אורך + איבר-איבר, בסדר-הכנסה כמו JSON.stringify) + ערכים — כלל-8.
import 'site-vocab.dart';

void _eqMap(String label, dynamic got, Map<String, dynamic> want) {
  final g = got as Map;
  final gKeys = g.keys.toList();
  final wKeys = want.keys.toList();
  if (gKeys.length != wKeys.length) {
    throw StateError('✗ $label: מספר-מפתחות ${gKeys.length} ≠ ${wKeys.length}');
  }
  for (var i = 0; i < wKeys.length; i++) {
    if (gKeys[i] != wKeys[i]) {
      throw StateError('✗ $label: מפתח[$i] ${gKeys[i]} ≠ ${wKeys[i]}');
    }
    final k = wKeys[i];
    if (g[k] != want[k]) {
      throw StateError("✗ $label: '$k' ⇒ ${g[k]} ≠ ${want[k]}");
    }
  }
}

void main() {
  // דוגמאות-חוזה 1–4 (זהות לבדיקת-ה-JS):
  _eqMap("(false,'he')", siteVocab(false, 'he'), {
    'heroCta': 'לתרומה עכשיו',
    'navCta': 'לתרומה ♡',
    'give': 'לתרומה ♡',
    'giveLabel': 'התרומה שלך',
    'commercial': false,
  });
  _eqMap("(false,'en')", siteVocab(false, 'en'), {
    'heroCta': 'Donate now',
    'navCta': 'Donate ♡',
    'give': 'Donate ♡',
    'giveLabel': 'Your gift',
    'commercial': false,
  });
  _eqMap("(true,'he')", siteVocab(true, 'he'), {
    'heroCta': 'צרו קשר',
    'navCta': 'צרו קשר',
    'give': 'צרו קשר',
    'giveLabel': 'הפנייה שלך',
    'commercial': true,
  });
  _eqMap("(true,'en')", siteVocab(true, 'en'), {
    'heroCta': 'Get in touch',
    'navCta': 'Contact',
    'give': 'Contact us',
    'giveLabel': 'Your request',
    'commercial': true,
  });
  // דוגמה 5: יידיש נופלת לעברית — זהה-ביט ל-(false,'he').
  final he = siteVocab(false, 'he') as Map;
  _eqMap("(false,'yi')≡(false,'he')", siteVocab(false, 'yi'),
      he.map((k, v) => MapEntry(k as String, v)));
  print('OK');
  print('✓ site-vocab (Dart): 5 דוגמאות-חוזה — ירוק');
}
